src/libraries/Common/src/System/Net/NetworkInformation/InterfaceInfoPal.Unix.cs

```diff
@@ -1,13 +1,65 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Runtime.InteropServices;
+using System.Text;
+
 namespace System.Net.NetworkInformation
 {
     internal static class InterfaceInfoPal
     {
+        private const int StackAllocationThreshold = 512;
+
         public static uint InterfaceNameToIndex(string interfaceName)
         {
             return Interop.Sys.InterfaceNameToIndex(interfaceName);
         }
+
+        public static unsafe uint InterfaceNameToIndex(ReadOnlySpan<char> interfaceName)
+        {
+            // Includes null terminator.
+            int bufferSize = Encoding.UTF8.GetByteCount(interfaceName) + 1;
```

Comment:
Technically this could overflow

Reply:
That's right. The scope always follows a `%`. The shortest IPv6 address which would pass IPv6AddressHelper.Parse is in the form "::%Long_Scope", so interfaceName will always have a maximum length of `int.MaxValue - 3`.

Reply:
Let's add an assert.